plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/gson_utils/EObjectSerializer.xtend

```diff
@@ -0,0 +1,173 @@
+                }
+                if (source.class === KTextImpl) {
+                    // Only KTexts have the additional calculatedTextBounds and calculatedTextLineWidths/Heights
+                    // properties.
+                    if (propertyHolder.hasProperty(KlighdProperties.CALCULATED_TEXT_BOUNDS)) {
+                        jsonObject.add("calculatedTextBounds", context.serialize(
+                            propertyHolder.getProperty(KlighdProperties.CALCULATED_TEXT_BOUNDS)))
+                    }
+                    if (propertyHolder.hasProperty(SprottyProperties.CALCULATED_TEXT_LINE_WIDTHS)) {
+                        jsonObject.add("calculatedTextLineWidths", context.serialize(
+                            propertyHolder.<float[]>getProperty(SprottyProperties.CALCULATED_TEXT_LINE_WIDTHS)))
+                    }
+                    if (propertyHolder.hasProperty(SprottyProperties.CALCULATED_TEXT_LINE_HEIGHTS)) {
+                        jsonObject.add("calculatedTextLineHeights", context.serialize(
+                            propertyHolder.<float[]>getProperty(SprottyProperties.CALCULATED_TEXT_LINE_HEIGHTS)))
+                    }
+                }
+            }
+            if (propertyHolder.hasProperty(KlighdProperties.TOOLTIP)) {
+                jsonObject.add("tooltip", context.serialize(
+                    propertyHolder.getProperty(KlighdProperties.TOOLTIP)))
+            }
+        }
+        return jsonObject
+    }
+    
+    /**
+     * Returns if a {@link Field} should be skipped during serialization.
+     * 
+     * @param f The field to be checked.
+     */
+    def shouldSkipField(Field f) {
+        return Modifier.isStatic(f.modifiers)
+            || KRenderingRef      .isAssignableFrom(f.declaringClass) && (f.getName().equals("rendering"))
+            || KRendering         .isAssignableFrom(f.declaringClass) && (f.getName().equals("placementData"))
+            || KContainerRendering.isAssignableFrom(f.declaringClass) && (f.getName().equals("childPlacement")) 
+    }
+    
+    /**
+     * Returns if a {@link Class} should be skipped during serialization.
+     * 
+     * @param c The class to be checked.
+     */
+    def shouldSkipClass(Class<? extends EObject> c) {
+        return KInsets.isAssignableFrom(c)
+    }
+    
+    /**
+     * Returns if a {@link Class} could have ambiguous types and need an additional "type" field to be added during
+     * serialization.
+     * 
+     * @param c The class to be checked.
+     */
+    def needsTypeField(Class<? extends EObject> c) {
+        return KRendering       .isAssignableFrom(c)
+            || KRenderingLibrary.isAssignableFrom(c)
+            || KStyle           .isAssignableFrom(c)
+            || KXPosition       .isAssignableFrom(c)
+            || KYPosition       .isAssignableFrom(c)
+    }
+    
+    // TODO: Special handling of default values
+    def specialHandling(Field f) {
+        return false
+    }
```

Comment:
This is for possibly handling some fields differently than other fields, as some numbers may be left out if they just contain the default value (which could therefore be restored on the client) or shortened. This might improve the performance slightly, but not too significantly so it was not done yet. There also is a ticket for this: [KIKEITH-34](https://rtsys.informatik.uni-kiel.de/jira/projects/KIKEITH/issues/KIKEITH-34)